src/datanode/src/datanode.rs

```diff
@@ -346,11 +345,22 @@ impl DatanodeBuilder {
         while let Some(table_value) = table_values.next().await {
             let table_value = table_value.context(GetMetadataSnafu)?;
             for region_number in table_value.regions {
+                // Augments region options with wal options.
+                // TODO(niebayes): fetch wal options map from region info.
+                // let wal_options_map = &table_value.region_info.wal_options_map;
+                let wal_options_map: HashMap<RegionNumber, EncodedWalOptions> = HashMap::default();
+                let wal_options = wal_options_map
+                    .get(&region_number)
+                    .cloned()
+                    .unwrap_or_default();
+                let mut region_options = table_value.region_info.region_options.clone();
+                region_options.extend(wal_options);
```

Comment:
Is it possible to store a json string instead of a hashmap for wal options? The EncodedHashMap is hard to deserialize.

Another approach is just to store the enum in the map.

Reply:
A `WalOptions` is simply a `String` now. See codes in another PR:
https://github.com/GreptimeTeam/greptimedb/blob/b2ec3d1c7126f3a98e8fd4f13764d9cf5c4f33d3/src/common/meta/src/wal.rs#L106-L126

Will update the deserialization for this PR soon.